drv/stm32h7-spi-server/build.rs

```diff
@@ -160,27 +182,31 @@ impl ToTokens for SpiConfig {
             .map(|(i, k)| (k, i))
             .collect();
 
+        // The svd2rust PAC can't decide whether acronyms are words, so we get
+        // to produce both identifiers.
+        let devname: syn::Ident =
+            syn::parse_str(&format!("SPI{}", self.controller)).unwrap();
+        let pname: syn::Ident =
+            syn::parse_str(&format!("Spi{}", self.controller)).unwrap();
+
         // We don't derive ToTokens for DeviceDescriptorConfig because it needs
         // extra knowledge (the mux_indices map) to do the conversion. Instead,
         // convert it here:
         let device_code = self.devices.values().map(|dev| {
             let mux_index = mux_indices[&dev.mux];
             let cs = &dev.cs;
+            let div: syn::Ident =
+                syn::parse_str(&format!("{:?}", dev.clock_divider)).unwrap();
```

Comment:
Not a fan on using Debug formatting in a non-debug usage like this.  I'd rather see Display implemented and used.

Reply:
I'd argue that Debug is a good choice here because we're trying to convert the enum back to its code representation, whereas Display is normally a human-readable output.

Reply:
https://doc.rust-lang.org/std/fmt/trait.Debug.html says `Debug should format the output in a programmer-facing, debugging context.` while https://doc.rust-lang.org/std/fmt/trait.Display.html says `Display is similar to Debug, but Display is for user-facing output, and so cannot be derived.` so neither is really correct for this situation.  strum's `AsRefStr` is probably the closest to "correct" and avoids the string allocation.

Reply:
Given that the type I'm formatting and its sole use are both within this same file, I don't see how this code is a problem. I'm basically trying to give a bit more feedback than just using String, but just using String would also work here.

Reply:
What if I add a large comment on the divider type noting that the names in the enum must match the names used in the PAC crate?

Reply:
I understand you are trying to get the name of an enum variant as a &str.  I found `&format!("{:?}", variant)` difficult to comprehend as doing that.  More generally, `Debug` is intended to be programmer-facing for debug contexts so using it in code generation feels wrong.